pkg/trustedcabundle/trustedcabundle.go

```diff
@@ -163,33 +163,42 @@ func IsTrustedCABundleUpdated(ctx context.Context, cli client.Client, dscInit *d
 }
 
 func ConfigureTrustedCABundle(ctx context.Context, cli client.Client, log logr.Logger, dscInit *dsci.DSCInitialization, managementStateChanged bool) error {
-	switch dscInit.Spec.TrustedCABundle.ManagementState {
-	case operatorv1.Managed:
-		log.Info("Trusted CA Bundle injection is set to `Managed` state. Reconciling to add/update cert configmaps")
-		istrustedCABundleUpdated, err := IsTrustedCABundleUpdated(ctx, cli, dscInit)
-		if err != nil {
-			return err
-		}
+	if dscInit.Spec.TrustedCABundle != nil {
+		switch dscInit.Spec.TrustedCABundle.ManagementState {
+		case operatorv1.Managed:
+			log.Info("Trusted CA Bundle injection is set to `Managed` state. Reconciling to add/update cert configmaps")
+			istrustedCABundleUpdated, err := IsTrustedCABundleUpdated(ctx, cli, dscInit)
+			if err != nil {
+				return err
+			}
 
-		if istrustedCABundleUpdated || managementStateChanged {
-			err = AddCABundleConfigMapInAllNamespaces(ctx, cli, dscInit)
+			if istrustedCABundleUpdated || managementStateChanged {
+				err = AddCABundleConfigMapInAllNamespaces(ctx, cli, dscInit)
+				if err != nil {
+					log.Error(err, "error adding configmap to all namespaces", "name", CAConfigMapName)
+					return err
+				}
+			}
+		case operatorv1.Removed:
+			log.Info("Trusted CA Bundle injection is set to `Removed` state. Reconciling to delete all cert configmaps")
+			err := RemoveCABundleConfigMapInAllNamespaces(ctx, cli)
 			if err != nil {
-				log.Error(err, "error adding configmap to all namespaces", "name", CAConfigMapName)
+				log.Error(err, "error deleting configmap from all namespaces", "name", CAConfigMapName)
 				return err
```

Comment:
Same principle as above

```suggestion
				fmt.Errorf("error deleting configmap %s from all namespaces", CAConfigMapName, err)
```